app/pages/challenges/components/challengeForm.tsx

```diff
@@ -20,24 +21,103 @@ const challengeSchema = z.object({
   participants: z.array(z.number()).optional(),
 });
 
+interface FriendOption {
+  id: number;
+  name: string;
+}
+
 interface ChallengeFormProps {
   onSubmit: (data: ChallengeRequest) => void;
   addError?: Error | null;
+  userId: number;
 }
 
 export const ChallengeForm: React.FC<ChallengeFormProps> = ({
   onSubmit,
   addError,
+  userId,
 }) => {
   const [formData, setFormData] = useState({
     title: "",
     description: "",
     goal: "",
     unit: "",
   });
-  const [friendIds, setFriendIds] = useState<string>("");
-  const [invitedFriends, setInvitedFriends] = useState<string[]>([]);
+  const [invitedFriends, setInvitedFriends] = useState<FriendOption[]>([]);
   const [errors, setErrors] = useState<Record<string, string>>({});
+  const [isSearchOpen, setIsSearchOpen] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const searchRef = useRef<HTMLDivElement>(null);
+
+  const [friends, setFriends] = useState<FriendOption[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadError, setLoadError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchFriends = async () => {
+      setIsLoading(true);
+      setLoadError(null);
+
+      try {
+        const response = await apis().friends.getFriends(userId);
+        if (response.status !== 200) {
+          throw new Error(`Error fetching friends: ${response.statusText}`);
+        }
+
+        const friendIds: number[] = await response.data;
+        const friendProfiles = await Promise.all(
+          friendIds.map(async (id) => {
+            try {
+              const profileResponse = await apis().user.getUserProfile(id);
```

Comment:
also use react query for getUserProfule , we have a hook already created. it profived you with loading , errors and caching

Reply:
https://www.youtube.com/watch?v=OrliU0e09io

Reply:
nice video 

to be fair you will proably use fetching in your job, but lets use react query here since it is a bit more tricky and more profesional